### PR TITLE
single arg ok for primerange

### LINKED ISSUE #19118: Generating first n prime numbers
In the `ntheory.generate` module, there seems to be no straightforward way to generate **the first _n_** primes, equivalent to `primerange(2, prime(n) + 1)`.

In contrast, the `primorial()` function supports computing the product of first _n_ primes as well as the product of primes less than or equal to _n_.

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

close #19118 

#### Brief description of what is fixed or changed

give `primerange` a single-arg option and document the use of `prime` to reference by index the desired number of primes to generate
```python
>>> primerange(prime(10))  # up to 10th
[2, 3, 5, 7, 11, 13, 17, 19, 23]
>>> primerange(prime(10)+1)  # including the 10th
[2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
```
#### Release Notes

* ntheory
  * `primerange` now accepts a single argument, a,  which gives the range [2, a)

### CODE REVIEW

File: sympy/ntheory/generate.py [hunk 1]
```diff
@@ -692,6 +696,7 @@ def primerange(a, b):
         See Also
         ========
 
+        prime : Return the ith prime
         nextprime : Return the ith prime greater than n
```
Comment (oscarbenjamin) [inline sympy/ntheory/generate.py:700]: What does `ith prime greater than n` mean?

Comment (smichr) [reply sympy/ntheory/generate.py:700]: It should read "return the next prime greater than i"

File: sympy/ntheory/generate.py [hunk 2]
```diff
@@ -692,6 +728,7 @@ def primerange(a, b):
         See Also
         ========
 
+        prime : Return the ith prime
```
Comment (smichr) [inline sympy/ntheory/generate.py:731]:
```suggestion
        prime : Return the nth prime
```

**Comment by smichr** [comment]
This adds functionality to function `primerange` that was already agreed upon for the sieve method with the same name.

**Pull request merged**